katharsis-core/src/main/java/io/katharsis/core/internal/utils/PropertyUtils.java

```diff
@@ -303,7 +303,7 @@ private void setPropertyValue(Object bean, String fieldName, Object value)
     }
 
     @SuppressWarnings("unchecked")
-    private Object prepareValue(Object value, Class<?> fieldClass) {
+    public static Object prepareValue(Object value, Class<?> fieldClass) {
```

Comment:
Where to start with this function?!

I don't know who wrote it but there is real potential for data loss here as a `List` can contain duplicates and upon putting it in a `Set` of any kind will remove them.

Next problem: Using a `XXXHashSet` of any kind assumes the elements in the `List` are hash/equals compatible. This assumption can easily fail and produce all kinds of weird results. Especially given the manual implementations of `hashCode/equals` in the code base

Basic of imperative flow control: at this line of code: `return new LinkedHashSet<>()` the function exists. That means that the conditional at 309 `else if` (itself and anti-pattern) is entirely redundant - a simple `if(...) return` works with no `else` nesting

This code is screaming data loss.